pkg/avx/collector.go

```diff
@@ -234,94 +362,73 @@ func (c collector) Collect(ch chan<- prometheus.Metric) {
 			ch <- prometheus.MustNewConstMetric(
 				descriptors[avxSwitchCountDesc],
 				prometheus.GaugeValue,
-				float64(binary.LittleEndian.Uint32(counters[idx_][0:])),
+				float64(counter_),
 				path,
-				fmt.Sprintf("%d", binary.LittleEndian.Uint64(cgroupid_[0:])))
+				fmt.Sprintf("%d", cgroupid_))
 
-			if err := c.bpfModule.LookupElement(c.allContextSwitchCounters, unsafe.Pointer(&cgroupid_[0]), unsafe.Pointer(&allCount)); err != nil {
-				log.Error("unable to find 'all' switch count: %+v", err)
-				return
-			}
-
-			if err := c.bpfModule.LookupElement(c.lastUpdateNs, unsafe.Pointer(&cgroupid_[0]), unsafe.Pointer(&lastUpdate)); err != nil {
+			if err := c.ebpf.Maps["last_update_ns"].Lookup(uint64(cgroupid_), &lastUpdate); err != nil {
 				log.Error("unable to find last update timestamp: %+v", err)
 				return
 			}
 
-			ch <- prometheus.MustNewConstMetric(
-				descriptors[allSwitchCountDesc],
-				prometheus.GaugeValue,
-				float64(allCount),
-				path)
-
 			ch <- prometheus.MustNewConstMetric(
 				descriptors[lastUpdateNsDesc],
 				prometheus.GaugeValue,
 				float64(nowNanoseconds()-lastUpdate),
 				path)
 
-		}(idx, cgroupid)
+		}(cgroupid, counter)
 	}
 
 	// We need to wait so that the response channel doesn't get closed.
 	wg.Wait()
 
-	_, _, err = c.moveAllElements(c.allContextSwitchCounters, unsafe.Sizeof(uint64(0)), unsafe.Sizeof(uint32(0)))
-	if err != nil {
-		log.Error("unable to delete elements of all_context_switch_count: %+v", err)
+	key = 0
+	iter = m.Iterate()
+	for iter.Next(&key, &val) {
```

Comment:
I get some C++ vibes here. :-)  Is it so that the deletion doesn't happen automatically when `m` goes out of scope?

Reply:
this deletes the map elements from the kernel memory, i.e., resets the counter values.